core/src/main/java/hudson/DNSMultiCast.java

```diff
@@ -77,16 +84,110 @@ public Object call() {
 
     public void close() {
         if (jmdns!=null) {
-//            try {
+            try {
                 jmdns.abort();
                 jmdns = null;
-//            } catch (final IOException e) {
-//                LOGGER.log(Level.WARNING,"Failed to close down JmDNS instance!",e);
-//            }
+            } catch (final IOException e) {
+                LOGGER.log(Level.WARNING,"Failed to close down JmDNS instance!",e);
+            }
         }
     }
 
     private static final Logger LOGGER = Logger.getLogger(DNSMultiCast.class.getName());
 
     public static boolean disabled = SystemProperties.getBoolean(DNSMultiCast.class.getName()+".disabled");
+
+    private static class JenkinsJmDNS extends JmDNSImpl {
```

Comment:
Would be nice to add Javadoc with the link to the original implementation and a summary of changes

Reply:
:+1:  I'll open a new PR to add the javadoc